org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/JavaLanguageServerPlugin.java

```diff
@@ -206,6 +218,38 @@ public IProxyService getProxyService() {
 		return null;
 	}
 
+	public String getProjectName(String typeFullyQualifiedName) throws CoreException {
+		SearchPattern pattern = SearchPattern.createPattern(typeFullyQualifiedName, IJavaSearchConstants.TYPE, IJavaSearchConstants.DECLARATIONS, SearchPattern.R_EXACT_MATCH);
+
+		IJavaSearchScope scope = SearchEngine.createWorkspaceScope();
+		ArrayList<String> projectNames = new ArrayList<>();
+
+		SearchRequestor requestor = new SearchRequestor() {
+			@Override
+			public void acceptSearchMatch(SearchMatch match) {
+				projectNames.add(((IJavaElement) match.getElement()).getJavaProject().getElementName());
+			}
+		};
+
+		SearchEngine searchEngine = new SearchEngine();
+		searchEngine.search(pattern, new SearchParticipant[] { SearchEngine.getDefaultSearchParticipant() }, scope, requestor, null /*progress monitor*/);
+		return projectNames.get(0);
+	}
+
+	public String computeClassPath(String projectName) throws CoreException {
+		IWorkspaceRoot root = ResourcesPlugin.getWorkspace().getRoot();
+		IProject project = root.getProject(projectName);
+		if (!project.isNatureEnabled("org.eclipse.jdt.core.javanature")) {
+			logError("Not a project with java nature.");
+			return null;
+		}
+		IJavaProject javaProject = JavaCore.create(project);
+		String[] classPathArray = JavaRuntime.computeDefaultRuntimeClassPath(javaProject);
+		String classPath = String.join(System.getProperty("path.separator"), classPathArray);
+		classPath = classPath.replaceAll("\\\\", "/");
```

Comment:
does computeClassPath return the common bin folder for java project where contains the compiled class file?